applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -262,7 +263,13 @@ def ComputeDeltaTime(self):
             raise Exception("::[MechanicalSolver]:: Time stepping not defined!")
 
     def GetComputingModelPart(self):
-        return self.main_model_part
+        computing_sub_model_part_name = self.settings["computing_sub_model_part_name"].GetString()
+        if computing_sub_model_part_name == "":
+            # if the user didn't specify a SubModelPart, then use the MainModelPart
+            return self.main_model_part
+        else:
+            computing_model_part_name = self.main_model_part.Name + "." + computing_sub_model_part_name
```

Comment:
I would have used the complete path as compute model part and would only check if it is a sub of main model part rather than going only one step down

Reply:
Now I see. You are right!